src/accelerate/accelerator.py

```diff
@@ -231,6 +255,70 @@ def prepare_model(self, model):
             model.forward = torch.cuda.amp.autocast()(model.forward)
         return model
 
+    def _prepare_deepspeed(self, *args):
+
+        ds_plugin = self.state.deepspeed_plugin
+        self.ds_config = ds_plugin.ds_config
+
+        bz = [obj.batch_size for obj in args if hasattr(obj, "batch_size")]
+        assert (
+            len(bz) > 0
+        ), "You must specify a training or evaluation dataloader in `accelerate.prepare()` when using DeepSpeed."
+        if len(bz) > 1:
+            logger.info(
+                "Since you passed both train & eval dataloader, `is_train_batch_min` will decide the `train_batch_size`"
+            )
+        batch_size_per_device = min(bz) if ds_plugin.is_train_batch_min else max(bz)
```

Comment:
Since we are computing it just after, can we log it? the log should be something like:
```
f"Since you passed both train and evaluation dataloader, `is_train_batch_min` (here {ds_plugin.is_train_batch_min} will decide the `train_batch_size` ({batch_size_per_device})."
```
This might need to be split in two lines as well to respect the 119 char limit.